### PR TITLE
Add Cucumber for testing, update gems

### PR DESCRIPTION
**Pull request by brucellino**
# Summary 

I've added [cucumber](cucumber.io) for behaviour-driven development, and [watir](watir.com) for the selenium driver. Some tests have been added to check for the presence of the proper login buttons.

I've also updated the gems to address security vulnerabilities.

----

### CODE REVIEW

**Comment by brucellino** [comment]
@enolfc @gwarf I would appreciate it if you could review this, just to spread the understanding around.

If there are any suggestions for making the codebase more readable, add them.
Also, if you can think of any behaviours that should be tested for on the forum, add them as extra issues.

File: tests/cucumber/features/login.feature [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Feature: Log in with CheckIn A person with an identity can log in with CheckIn
+  Scenario: The login button is present
```
Comment (enolfc) [inline tests/cucumber/features/login.feature]: I'm curious on how do you plan to test other scenarios here, mainly, testing an actual login with Check-in, is that feasible at all?

Comment (gwarf) [reply tests/cucumber/features/login.feature]: As long as we have proper credentials with Selenium you can do whatever you want. We could easily create a google account for a test user and used it access discourse.

Comment (brucellino) [reply tests/cucumber/features/login.feature]: Exactly. Discussed this a few times with @nliampotis

**Comment by brucellino** [comment]
Thanks for the feedback @gwarf ! I'm not sure how easy it will be to use a social login, but there should be a few "test" accounts in checkin for this kind of thing. I might ask to do this in a subsequent PR

**Pull request merged**